python_modules/dagster/dagster/core/execution.py

```diff
@@ -583,33 +588,62 @@ def construct_pipeline_execution_context(
     )
 
 
-def _create_loggers(run_config, execution_context):
+def _create_loggers(environment_config, run_config, pipeline_def):
+    check.inst_param(environment_config, 'environment_config', EnvironmentConfig)
     check.inst_param(run_config, 'run_config', RunConfig)
-    check.inst_param(execution_context, 'execution_context', ExecutionContext)
+    check.inst_param(pipeline_def, 'pipeline_def', PipelineDefinition)
+
+    loggers = []
+    for logger_key, logger_def in pipeline_def.loggers.items() or default_loggers().items():
+        if logger_key in environment_config.loggers:
+            init_logger_context = InitLoggerContext(
+                environment_config.context,
```

Comment:
minor style thing, but I generally try to eschew reusing variables (in this case init_logger_context). Variables should ideally be "write once". In this case I think you can just get rid of all of them and pass them in directly

Reply:
Totally agree